ranger/container/history.py

```diff
@@ -5,30 +5,32 @@
 
 from __future__ import (absolute_import, division, print_function)
 
+from ranger.core.shared import SettingsAware
+
 
 class HistoryEmptyException(Exception):
     pass
 
 
-class History(object):
+class History(SettingsAware, object):
 
-    def __init__(self, maxlen=None, unique=True):
-        maxlen = float('inf') if maxlen is None else maxlen
-        if isinstance(maxlen, History):
-            self.history = list(maxlen.history)
-            self.index = maxlen.index
-            self.maxlen = maxlen.maxlen
-            self.unique = maxlen.unique
+    def __init__(self, obj=None, maxlen=None, unique=True):
```

Comment:
I don't like adding new arguments, especially not anywhere but the back of the list. If any calls used positional arguments they might now be broken.

Reply:
"AttributeError: 'History' object has no attribute 'settings'", because new History get settings dependence.
May be we can change test for this, but I don't know.

Reply:
I checked history calls, it's ok.
But new arg more logical, is not it?